velox/functions/prestosql/ArrayFunctions.h

```diff
@@ -729,6 +732,120 @@ inline void checkIndexArrayTrim(int64_t size, int64_t arraySize) {
   }
 }
 
+/// This class implements the array_top_n function.
+///
+/// DEFINITION:
+/// array_top_n(array(T), int) -> array(T)
+/// Returns the top n elements of the array in descending order.
+template <typename T>
+struct ArrayTopNFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  // Definition for primitives.
+  template <typename TReturn, typename TInput>
+  FOLLY_ALWAYS_INLINE void
+  call(TReturn& result, const TInput& array, int64_t n) {
+    VELOX_CHECK(
+        n >= 0, fmt::format("Parameter n: {} to ARRAY_TOP_N is negative", n));
+
+    // Define min-heap to store the top n elements.
+    std::priority_queue<
+        typename TInput::element_t,
+        std::vector<typename TInput::element_t>,
+        std::greater<>>
+        minHeap;
+
+    // Iterate through the array and push elements to the min-heap.
+    int numNull = 0;
+    for (const auto& item : array) {
+      if (item.has_value()) {
+        minHeap.push(item.value());
```

Comment:
can you do a check on the value against the top() in the minHeap before pushing it on the heap (both here and in the generic implementation). In the current implementation if you always add it then it would end up doing many more comparisons and switches to balance the heap which can be wasteful